torchtune/models/clip/_transforms.py

```diff
@@ -5,24 +5,112 @@
 # LICENSE file in the root directory of this source tree.
 
 import logging
+import math
 from typing import Any, List, Mapping, Optional, Tuple
 
 import torch
-import torchvision
 from PIL import Image
 
 from torchtune.modules.transforms import (
     find_supported_resolutions,
     get_canvas_best_fit,
-    resize_with_pad,
-    tile_crop,
+    TileCrop,
 )
+from torchvision.transforms import v2
+from torchvision.transforms._functional_tensor import resize
 
 from torchvision.transforms.v2 import functional as F
 
 logger = logging.getLogger(__name__)
 
 
+class CLIPImageTransformCore(torch.nn.Module):
```

Comment:
Just to confirm after our discussion today, the idea is to fully make `CLIPImageTransform` exportable, but we are not there yet, right? If so, do we intend to land this as an intermediate version first? The reason I ask is because with this PR some of our utilities like `resize_with_pad` will no longer have any callsites in the repo, and I want to know whether this is expected to be the final state or just a temporary one.

Reply:
Yeah - ideally we can fully export `CLIPImageTransform` but right now that would require more support on export side. So I think we can remove `resize_with_pad` and its tests.